firewood/src/merkle/node.rs

```diff
@@ -360,24 +374,46 @@ impl Storable for Node {
             None
         };
 
+        let is_encoded_longer_than_hash_len =
+            if attrs.contains(NodeAttributes::IS_ENCODED_BIG_VALID) {
+                Some(false)
+            } else if attrs.contains(NodeAttributes::LONG) {
+                Some(true)
+            } else {
+                None
+            };
+
+        let meta_raw = mem
+            .get_view(offset, 1_u64)
+            .ok_or(ShaleError::InvalidCacheView {
+                offset,
+                size: 1_u64,
+            })?;
+
+        offset += 1;
+
         #[allow(clippy::indexing_slicing)]
-        match meta_raw.as_deref()[TRIE_HASH_LEN + 1].try_into()? {
+        match meta_raw.as_deref()[0].try_into()? {
```

Comment:
If you add the `NodeTypeId` to `Meta`, you can leave this code as is. The idea here was actually to create the `Meta` struct first then use its properties... not sure why I didn't do that. Regardless, dealing with all the meta-data should be done in one atomic step. There should only be one read from `mem`, then `mem` should be passed to the node deserialization.